18444 Build style guide with gulp-livingstyleguide

myabc opened the pull request:
:warning: **This PR doesn't remove building of the style guide with the assets pipeline.**

https://community.openproject.org/work_packages/18444

myabc committed:
Add gulp-livingstyleguide

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Make gulp sass task use Bower-installed Bourbon

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add gulp task to build style guide

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add gulp task to copy fonts to frontend/public

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Move express server script to frontend root

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Config Bourbon w/asset pipeline only if available

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Make default gulp task build style guide

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Tell gulp to watch LSG markdown files

Signed-off-by: Alex Coles <alex@alexbcoles.com>

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://myabc-feature-18444-gulp-livingstyleguide.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://myabc-feature-18444-gulp-livingstyleguide.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

myabc committed:
Fix missing semicolon in Gulpfile

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix typo breaking Protractor mock API

Broken in 572a0dc0.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Use same output dir for styleguide/fonts/sass tasks

This patch should also fix loading of fonts when using the Express
development server.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Mention gulp dev task in DEVELOP

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Move Angular templates under frontend/app/templates

We do not serve these templates directly. Rather Webpack always bundles them.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Use custom Jade template w/gulp-livingstyleguide

This achieves design parity with the asset pipeline-built style guide.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'dev' into feature/18444-gulp-livingstyleguide

floriank commented:
:+1:
